conans/client/api/conan_api.py

```diff
@@ -90,19 +166,79 @@ def user_update(self, user_name, user_pasword):
         return {}
 
     @api_method
-    def search_recipes(self, query, remote_patterns=None, local_cache=False):
-        remote = None
-        if remote_patterns is not None and len(remote_patterns) > 0:
-            remote = remote_patterns[0].replace("*", "remote")
+    def search_local_recipes(self, query):
+        data = {
+            "remote": None,
+            "error": False,
+            "results": None
+        }
+
+        remotes = self.app.cache.registry.load_remotes()
+        search = Search(self.app.cache, self.app.remote_manager, remotes)
+
+        try:
+            references = search.search_local_recipes(query)
+        except ConanException as exc:
+            data["error"] = True
+            raise
+
+        results = []
+        for reference in references:
+            result = {
+                "name": reference.name,
+                "id": repr(reference)
+            }
+            results.append(result)
+
+        data["results"] = results
+        return data
+
+    @api_method
+    def search_remote_recipes(self, remote, query):
+        data = {
+            "remote": remote.name,
+            "error": False,
+            "results": None
+        }
+
+        remotes = self.app.cache.registry.load_remotes()
+        search = Search(self.app.cache, self.app.remote_manager, remotes)
+
+        results = []
+        remote_references = search.search_remote_recipes(query, remote.name)
+        for remote_name, references in remote_references.items():
+            for reference in references:
+                result = {
+                    "name": reference.name,
+                    "id": repr(reference)
+                }
+                results.append(result)
+
+        data["results"] = results
+        return data
+
+    @api_method
```

Comment:
Probably this is not an ``api_method``, but an auxiliary one. ``api_methods`` are those that match the Conan command, and need to have a "fresh state". Something to be discussed for the api, not this PR.